### PR TITLE
Infinifilter implementation

### PR DESCRIPTION
**Pull request by ianopolous**

### CODE REVIEW

File: src/main/java/org/peergos/blockstore/CidInfiniFilter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import io.ipfs.cid.*;
+import org.peergos.blockstore.filters.*;
+
+import java.util.*;
+
+public class CidInfiniFilter implements Filter {
+
+    private final ChainedInfiniFilter filter;
+
+    public CidInfiniFilter(ChainedInfiniFilter filter) {
```
Comment (kevodwyer) [inline src/main/java/org/peergos/blockstore/CidInfiniFilter.java:12]: can this be private?

File: src/main/java/org/peergos/blockstore/filters/ChainedInfiniFilter.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	int count_until_replacing_former = 0;
+	int count_until_expanding_former = 0;
+	int former_phase = 0;
+	
+	public ChainedInfiniFilter(int power_of_two, int bits_per_entry) {
+		super(power_of_two, bits_per_entry);
+		older_filters = new ArrayList<InfiniFilter>();
+	}
+	
+	void handle_empty_fingerprint(long bucket_index, QuotientFilter current) {
```
Comment (kevodwyer) [inline src/main/java/org/peergos/blockstore/filters/ChainedInfiniFilter.java:53]: missing acccess permission

File: src/main/java/org/peergos/blockstore/filters/HashFunctions.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	public static int cryptographic_hash(String key) {
+		BigInteger number = new BigInteger("0");
+		
+		try {
+			number = new BigInteger(1, HashFunctions.getSHA(key));
+		} catch (NoSuchAlgorithmException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (kevodwyer) [inline src/main/java/org/peergos/blockstore/filters/HashFunctions.java:35]: should throw runtime exception

**Review by kevodwyer — COMMENTED** [review]
Are we ok with leaving the code style as is?

**Comment by ianopolous** [comment]
Yep, I haven't made any changes to the imported code.

File: src/main/java/org/peergos/blockstore/filters/HashFunctions.java (hunk 3 above)
Comment (ianopolous) [reply src/main/java/org/peergos/blockstore/filters/HashFunctions.java:35]: It's unreachable code.

**Review by kevodwyer — APPROVED** [review]

**Pull request merged**